modules/tracking/src/gtrTracker.cpp

```diff
@@ -75,7 +75,9 @@ class TrackerGOTURNModel : public TrackerModel{
 public:
     TrackerGOTURNModel(TrackerGOTURN::Params){}
     Rect2d getBoundingBox(){ return boundingBox_; }
-    void setBoudingBox(Rect2d boundingBox){ boundingBox_ = boundingBox; }
+    void setBoudingBox(Rect2d boundingBox) {
+        boundingBox_ = boundingBox & Rect2d(Point(0, 0), image_.size());;
```

Comment:
Please check that `image_` is not empty:
```
CV_Check(image_, !image_.empty(), "Set image first");
```

Reply:
In ```setBoudningBox()``` ?

Reply:
Right before 
```cpp
boundingBox_ = boundingBox & Rect2d(Point(0, 0), image_.size());;
```
because in case of empty image `mage_.size()` is`(0, 0)`

Reply:
OK

Reply:
I tried to rebuild the library with these changes:
```cpp
    void setBoudingBox(Rect2d boundingBox) {
        CV_Check(image_, !image_.empty(), "Set image first");
        boundingBox_ = boundingBox & Rect2d(Point(0, 0), image_.size());
    }
```
but got an error:
```
from /home/ilyaelizarov/opencv_contrib/modules/tracking/src/gtrTracker.cpp:42:
/home/ilyaelizarov/opencv_contrib/modules/tracking/src/gtrTracker.cpp: In member function ‘void cv::gtr::TrackerGOTURNModel::setBoudingBox(cv::Rect2d)’:
/home/ilyaelizarov/opencv/modules/core/include/opencv2/core/check.hpp:105:78: error: no matching function for call to ‘check_failed_auto(cv::Mat&, const cv::detail::CheckContext&)’
         cv::detail::check_failed_ ## type((v), CV__CHECK_LOCATION_VARNAME(id));
```

Reply:
Better 
```cpp
if (image_.empty())
    CV_Error(Error::StsInternal, "Set image first");
```

Reply:
It works, thank you!